metadata-ingestion/src/datahub/ingestion/source/iceberg/iceberg_common.py

```diff
@@ -50,32 +50,18 @@ class IcebergProfilingConfig(ConfigModel):
     # include_field_sample_values: bool = True
 
 
-class IcebergCatalogConfig(ConfigModel):
-    """
-    Iceberg catalog config.
-
-    https://py.iceberg.apache.org/configuration/
-    """
-
-    name: str = Field(
-        default="default",
-        description="Name of catalog",
-    )
-    type: str = Field(
-        description="Type of catalog.  See [PyIceberg](https://py.iceberg.apache.org/configuration/) for list of possible values.",
-    )
-    config: Dict[str, str] = Field(
-        description="Catalog specific configuration.  See [PyIceberg documentation](https://py.iceberg.apache.org/configuration/) for details.",
-    )
-
-
 class IcebergSourceConfig(StatefulIngestionConfigBase, DatasetSourceConfigMixin):
     # Override the stateful_ingestion config param with the Iceberg custom stateful ingestion config in the IcebergSourceConfig
     stateful_ingestion: Optional[StatefulStaleMetadataRemovalConfig] = Field(
         default=None, description="Iceberg Stateful Ingestion Config."
     )
-    catalog: IcebergCatalogConfig = Field(
-        description="Catalog configuration where to find Iceberg tables.  See [pyiceberg's catalog configuration details](https://py.iceberg.apache.org/configuration/).",
+    # The catalog configuration is using a dictionary to be open and flexible.  All the keys and values are handled by pyiceberg.  This will future-proof any configuration change done by pyiceberg.
+    # catalog: Dict[str, Dict[str, str]] = Field(
+    #     description="Catalog configuration where to find Iceberg tables.  Only one catalog specification is supported.  The format is the same as [pyiceberg's catalog configuration](https://py.iceberg.apache.org/configuration/), where the catalog name is specified as the object name and attributes are set as key-value pairs.",
+    # )
+    # `catalog` field to accept `Any` to handle both new and deprecated formats.  Once deprecated format is not supported, we can remove this field and use the above `catalog` field.
+    catalog: Any = Field(
```

Comment:
actually I think the type here should be only `Dict[str, Dict[str, str]]` as you had previously

the validator will automatically convert the old format to the new one, but we don't want to show the old type in the docs, and we want the correct type annotation so that mypy can check our usages of the configs

Reply:
You are absolutely right!  I made that change before implementing the validator.